lld/ELF/Writer.cpp

```diff
@@ -1124,11 +1127,15 @@ template <class ELFT> void Writer<ELFT>::setReservedSymbolSections() {
   }
 
   if (last) {
-    // _edata points to the end of the last mapped initialized section.
+    // _edata points to the end of the last mapped initialized section before
```

Comment:
Apart from the comment about .ldata may be after _edata, is this something we had wrong prior to this patch?

If so it could be worth splitting out into a separate patch so we can identify/test it separately. Ideally we'd want a test case for each of the conditions below.

Reply:
+1

Reply:
I find that `_etext` needs adjustment and just pushed a commit to fix it.

Hmm. I think splitting this part into a separate PR might be difficult due to the inflexible GitHub PR mechanism.

The test difference is in `lld/test/ELF/x86-64-section-layout.s` below:
```patch
-# CHECK-NEXT: 0000000000205b0a 0 NOTYPE GLOBAL DEFAULT [[#]] (.ldata2) _edata
+# CHECK-NEXT: 0000000000203307 0 NOTYPE GLOBAL DEFAULT [[#]] (.data) _edata
```